src/Graphics/src/Graphics/Platforms/Windows/PlatformImage.cs

```diff
@@ -153,10 +154,38 @@ public IImage ToImage(int width, int height, float scale = 1f)
 		public static IImage FromStream(Stream stream, ImageFormat format = ImageFormat.Png)
 		{
 			var creator = PlatformGraphicsService.Creator;
+			
 			if (creator == null)
+			{
 				throw new Exception("No resource creator has been registered globally or for this thread.");
+			}
+
+			CanvasBitmap bitmap;
+
+			if (stream.CanSeek)
+			{
+				global::Windows.Foundation.IAsyncOperation<CanvasBitmap> bitmapAsync = CanvasBitmap.LoadAsync(creator, stream.AsRandomAccessStream());
+				bitmap = bitmapAsync.AsTask().GetAwaiter().GetResult();
+			}
+			else
+			{
+				byte[] bytes = ArrayPool<byte>.Shared.Rent(100 * 1024); // 100 kB.
+
+				try
+				{
+					using MemoryStream memoryStream = new(bytes);
+					stream.CopyTo(memoryStream);
+					memoryStream.Seek(0, SeekOrigin.Begin);
+
+					global::Windows.Foundation.IAsyncOperation<CanvasBitmap> bitmapAsync = CanvasBitmap.LoadAsync(creator, memoryStream.AsRandomAccessStream());
+					bitmap = bitmapAsync.AsTask().GetAwaiter().GetResult();
```

Comment:
This is certainly a bit better than just doing `new MemoryStream()` each time, and probably "good enough". If non-seekable streams didn't work at all before, I would go with this.

The only problematic part now:
1. Any image _larger_ than 100kb
2. `MemoryStream` will throw away the `byte[]` and create a new, larger one
3. `MemoryStream` might do this multiple times as it continues to read. I believe the behavior is it will keep creating new `byte[]` 2x larger each try.

Reply:
The only way I know is to recycle `MemoryStream` like we do here:
* https://github.com/dotnet/android-tools/blob/main/src/Microsoft.Android.Build.BaseTasks/MemoryStreamPool.cs

But this loses the benefit that `ArrayPool.Shared.Rent()` also shares the array pool with the BCL.

If I find another reason to bring over our `MemoryStreamPool`, I'll update this in the future to use it.

Reply:
Is this an option we could evaluate?
https://github.com/microsoft/Microsoft.IO.RecyclableMemoryStream

Reply:
I saw that one (it showed up a few years after the Xamarin.Android one), but I haven't had a chance to try it.

I wonder if some of their default settings are suitable for mobile:
* https://github.com/microsoft/Microsoft.IO.RecyclableMemoryStream/blob/e29a28387da9018fa9605a1dcb3f7a0435aa9974/src/RecyclableMemoryStreamManager.cs#L52

Either one would probably work for this case on Windows.

Reply:
The issue is on windows here, so I was thinking about using this on `WINDOWS` only.

Reply:
For the record: https://github.com/dotnet/runtime/issues/22428#issuecomment-1478585251